pkg/apis/core/validation/validation.go

```diff
@@ -1243,13 +1243,33 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 		return allErrs
 	}
 
-	privileged := container.SecurityContext != nil && container.SecurityContext.Privileged != nil && *container.SecurityContext.Privileged
-	if *mountPropagation == core.MountPropagationBidirectional && !privileged {
-		allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers"))
+	if *mountPropagation == core.MountPropagationBidirectional {
+		// additional permissions required for mount functionality
+		if !hasBidirectionalMountPropagationPermissions(container.SecurityContext) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers and containers with the 'SYS_ADMIN' capability"))
+		}
 	}
 	return allErrs
 }
 
+func hasBidirectionalMountPropagationPermissions(securityContext *core.SecurityContext) bool {
+	if securityContext == nil {
+		// No security context configured
+		return false
+	}
+
+	// check container for required capability
+	if securityContext.Capabilities != nil {
+		for _, capability := range securityContext.Capabilities.Add {
+			if capability == "SYS_ADMIN" {
```

Comment:
is this comparison accurate? I can never remember whether capabilities are case-insensitive or not, or can be prefixed with `CAP_` or not

Reply:
⚠️ It seems that lowercase and uppercase both add the correct capability, but the `cap_` prefix does not work.

Tested existing behavior on k8s 1.27.2:
1. Create pod using following manifest, rotating uncommented capability each time
    ```yaml
    apiVersion: v1
    kind: Pod
    metadata:
      name: test
      annotations:
        container.apparmor.security.beta.kubernetes.io/alpine: unconfined
    spec:
      containers:
      - name: alpine
        image: alpine
        command: ['sh', '-c', 'apk add -U libcap && capsh --print']
        securityContext:
          capabilities:
            add:
            - "SYS_ADMIN"
            # - "sys_admin"
            # - "CAP_SYS_ADMIN"
            # - "cap_sys_admin"
            # - "INVALID_CAPABILITY"
    ```
2. Check logs of resulting pod whether container has added capability

Following behavior observed when trying to add `sys_admin` capability to container:
- valid, uppercase capability: `SYS_ADMIN`
    - pod scheduled successfully
    - container has `cap_sys_admin` capability
        ```
        Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_sys_admin,cap_mknod,cap_audit_write,cap_setfcap=ep
        ```
- valid, lowercase capability: `sys_admin`
    - pod scheduled successfully
    - container has `cap_sys_admin` capability
        ```
        Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_sys_admin,cap_mknod,cap_audit_write,cap_setfcap=ep
        ```
- valid, uppercase capability with `CAP` prefix: `CAP_SYS_ADMIN`
    - pod scheduled successfully
    - container DOES NOT have `cap_sys_admin` capability
        ```
        Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap=ep
        ```
- valid, lowercase capability with `cap` prefix: `cap_sys_admin`
    - pod scheduled successfully
    - container DOES NOT have `cap_sys_admin` capability
        ```
        Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap=ep
        ```
- invalid capability: `INVALID_CAPABILITY`
    - pod scheduled successfully

I will open an issue with these findings and try to improve/abstract the capability handling in the code base, if these findings look fine to you.